src/Microsoft.ML.Data/Transforms/RowShufflingTransformer.cs

```diff
@@ -652,14 +652,14 @@ protected override bool MoveNextCore()
                     // We should let the producer know it can give us more stuff.
                     // It is possible for int values to be sent beyond the
                     // end of the sentinel, but we suppose this is irrelevant.
-                    PostAssert(_toProduce, _deadCount);
+                    PostAssert(_toProduceChannel, _deadCount);
                     _deadCount = 0;
                 }
 
                 while (_liveCount < _poolRows && !_doneConsuming)
                 {
                     // We are under capacity. Try to get some more.
-                    int got = _toConsume.Receive();
+                    _toConsumeChannel.Reader.TryRead(out int got);
```

Comment:
You are ignoring the return value of `TryRead` here. It can return `false` if there is nothing to read right now.